### PR TITLE
chore(kafka): switch to another error crate: thiserror

### PR DESCRIPTION
**Pull request by olksdr**
The crate we use for error handling `failure` is deprecated and outdated. `thiserror` is a good, near drop-in replacement for `#[derive(Fail)]`.

#skip-changelog

### CODE REVIEW

File: relay-kafka/src/lib.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@
 //!     builder = builder.add_kafka_topic_config(KafkaTopic::Events, &kafka_config_events).unwrap();
 //!
 //!     // And add potentially another topic config.
-//!     builder.add_kafka_topic_config(&kafka_config_metrics)?;
+//!     builder = builder.add_kafka_topic_config(KafkaTopic::Transactions, &kafka_config_metrics).unwrap();
```
Comment (iker-barriocanal) [inline relay-kafka/src/lib.rs:19]: Is this change related to the PR?

**Review by iker-barriocanal — APPROVED** [review]
`thiserror` has the same [licensing](https://github.com/dtolnay/thiserror) (MIT or Apache) as [`serde_json`](https://github.com/serde-rs/json), so there shouldn't be any issues regarding that.

File: relay-kafka/src/lib.rs (hunk 1 above)
Comment (olksdr) [reply relay-kafka/src/lib.rs:19]: This is just to make the doc consistent, and use the proper params to to show the use-case.

**Review by jan-auer — APPROVED** [review]
:shipit:
